### DISCUSSION TITLE
Recreating i3info

### DISCUSSION THREAD

**Discussion by Technicus** [budlabs/i3ass#167] [Show and tell]
Hey budlabs,

Your work is excellent and inspiring.
I have tried to recreate your super top secret i3info program.
You indicate that you may or may not release your version.
My version has some problems, and I am looking for help making it reliable.
It is here [i3_info](https://github.com/Technicus/i3_info).

Thanks for making i3ass and all your awesome videos!

**Comment by budRich** [comment]
Awesome! I see in your readme, that a parsing error occurs. It is most likely a nasty window title that breaks it. I would do this:  

1. find a way to control window titles. one easy way is to create html files with just `<head><title>My nice title</title></head>` and open them in a browser.
2. Intentionally try to crash the parser by making a bad title. (include single/doublequotes backslash e.t.c.)
3. file issue @ https://github.com/altdesktop/i3ipc-python/issues

I wrote my version in bash, and made a quite massive regex to parse the json. I have had some issues, but managed to perfect that regex over time, and now its been long since it had problems. And if it would have problems, it doesn't crash, it simply doesn't update the info.

I guess that is a solution as well, if there is some way to keep the script going even if `i3.get_tree()` fails.

My script, [i3king](https://github.com/budlabs/i3ass/tree/next/src/i3king), uses a similar method as my secret i3info script, [this part](https://github.com/budlabs/i3ass/blob/e78979733bdefa5ec0410ad9ddaf2a73066d920a/src/i3king/i3king#L78) is what takes the regex created [here](https://github.com/budlabs/i3ass/blob/next/src/i3king/func/make_regex.sh) and "transforms" the json to a bash array.

I have been wanting to make a GTK variant of i3info.

I hope you will resolve the issue. Have a nice day!

**Reply by Technicus** [reply]
Wow,

Your work is very thorough. Really top notch professional grade development. You set a bar for me to achieve.

A solution for the problem I am having is to switch from `tree = await i3.get_tree()` to `workspace = check_output("i3-msg -t get_workspaces | jq '.[] | select(.focused==true).name' | cut -d'\"' -f2", shell=True)`. The error no longer happens. I went onto the i3 IRC channel and someone suggested that it was not the best solution, but it works for now.

I am still new at this and want to move on to other things. I can come back to it after I get a better understanding. There is so much to setup, and my patients for this problem has expired.

Originally I was going to add curses to my version. A GTK version might be interesting. It is funny . . . I have only been learning i3 for about a month and I am still "ricing" my configuration. I have been following your series but I stopped after I found out that you didn't release your i3info to make my own.

It is a cool utility and I really want it, but I am glad you did not release it. It inspired me to make my own, and in doing so I had to learn a few other things. It has been somewhat of a struggle, and sort of discouraging at times, but I have learned a few concepts I can implement as I continue to rice my config.

Thanks budRich!

Also, if there is anything I can do to help you with development | testing let me know and I will do what I can.
